mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/CassandraMailboxTestRunner.java

```diff
@@ -0,0 +1,447 @@
+import org.apache.james.mailbox.cassandra.mail.CassandraModSeqProviderContract;
+import org.apache.james.mailbox.cassandra.mail.CassandraUidProvider;
+import org.apache.james.mailbox.cassandra.mail.CassandraUidProviderContract;
+import org.apache.james.mailbox.cassandra.mail.CassandraUserMailboxRightsDAO;
+import org.apache.james.mailbox.cassandra.mail.CassandraUserMailboxRightsDAOContract;
+import org.apache.james.mailbox.cassandra.mail.MailboxAggregateModule;
+import org.apache.james.mailbox.cassandra.mail.migration.AttachmentMessageIdCreation;
+import org.apache.james.mailbox.cassandra.mail.migration.AttachmentMessageIdCreationContract;
+import org.apache.james.mailbox.cassandra.mail.migration.AttachmentV2Migration;
+import org.apache.james.mailbox.cassandra.mail.migration.AttachmentV2MigrationContract;
+import org.apache.james.mailbox.cassandra.mail.migration.CassandraMailboxMapperFallbackContract;
+import org.apache.james.mailbox.cassandra.modules.CassandraQuotaModule;
+import org.apache.james.mailbox.cassandra.modules.CassandraRegistrationModule;
+import org.apache.james.mailbox.cassandra.quota.CassandraCurrentQuotaManager;
+import org.apache.james.mailbox.cassandra.quota.CassandraGlobalMaxQuotaDao;
+import org.apache.james.mailbox.cassandra.quota.CassandraPerDomainMaxQuotaDao;
+import org.apache.james.mailbox.cassandra.quota.CassandraPerUserMaxQuotaDao;
+import org.apache.james.mailbox.cassandra.quota.CassandraPerUserMaxQuotaManager;
+import org.apache.james.mailbox.cassandra.user.CassandraSubscriptionMapper;
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.mailbox.model.MessageId;
+import org.apache.james.mailbox.quota.MaxQuotaManager;
+import org.apache.james.mailbox.store.mail.AnnotationMapper;
+import org.apache.james.mailbox.store.mail.AttachmentMapper;
+import org.apache.james.mailbox.store.mail.model.AnnotationMapperContract;
+import org.apache.james.mailbox.store.mail.model.AttachmentMapperContract;
+import org.apache.james.mailbox.store.mail.model.Mailbox;
+import org.apache.james.mailbox.store.mail.model.MailboxMapperACLContract;
+import org.apache.james.mailbox.store.mail.model.impl.SimpleMailbox;
+import org.apache.james.mailbox.store.quota.GenericMaxQuotaManagerTest;
+import org.apache.james.mailbox.store.quota.StoreCurrentQuotaManager;
+import org.apache.james.mailbox.store.quota.StoreCurrentQuotaManagerContract;
+import org.apache.james.mailbox.store.user.SubscriptionMapper;
+import org.apache.james.mailbox.store.user.SubscriptionMapperContract;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(DockerCassandraExtension.class)
+class CassandraMailboxTestRunner implements CassandraUidProviderContract, CassandraModSeqProviderContract, CassandraMailboxDAOContract ,
```

Comment:
Wow, that's a huge drawback in term of maintenance, what is the gain of this exact commit ?

Reply:
That's a 60% reduction in Cassandra mailbox testing time... As eplained in above comments, it went from 13min29 to 5min20.